### DISCUSSION TITLE
Composite functions have two differents implementation styles. Why?

### DISCUSSION THREAD

**Discussion by Gounlaf** [dlemstra/Magick.NET#1559] [General] [answered]
:wave: 

This one use `TemporaryDefines`, relying on the the `using` keywork / disposable of `TemporaryDefines` to remove the artifact.

https://github.com/dlemstra/Magick.NET/blob/c022bb00a2df3928ca6d8ce8c1a2445784337b2c/src/Magick.NET/MagickImage.cs#L2044-L2052

This one doesn't

https://github.com/dlemstra/Magick.NET/blob/c022bb00a2df3928ca6d8ce8c1a2445784337b2c/src/Magick.NET/MagickImage.cs#L2189-L2205

The only particular reason I see is that `CompositeGravity` would allow en empty (but not null) `args` string, and all other methods using the `TemporaryDefines` would not?

https://github.com/dlemstra/Magick.NET/blob/c022bb00a2df3928ca6d8ce8c1a2445784337b2c/src/Magick.NET/Helpers/TemporaryDefines.cs#L28-L35

Thanks for the clarifications.

Regards.

**Comment by dlemstra** [comment, marked as answer]
That is code that has not yet been changed to use `TemporaryDefines` instead. I just pushed a patch to fix this. And as always thanks for reporting all these things.

**Reply by Gounlaf** [reply]
My pleasure 🙂
